src/lib.rs

```diff
@@ -95,26 +95,26 @@ fn reject_self_copies(sources: &[PathBuf], dest: &Path) -> Result<()> {
         dest.to_path_buf()
     };
 
-    // We use `fs::metadata` for `ancestor_inos` since we do the exact same thing regardless of
+    // We use `fs::metadata` for `ancestor_dev_inos` since we do the exact same thing regardless of
     // whether `dest` is a directory or a symlink pointing to one.
-    let ancestor_inos = dest
+    let ancestor_dev_inos = dest
         .ancestors()
-        .map(|ancestor| fs::metadata(ancestor).map(|meta| meta.ino()));
+        .map(|ancestor| fs::metadata(ancestor).map(|meta| format!("{}/{}", meta.dev().to_string(), meta.ino().to_string())));
 
-    // In contrast, we use `fs::symlink_metadata` for `source_inos` because we copy the symlinks
+    // In contrast, we use `fs::symlink_metadata` for `source_dev_inos` because we copy the symlinks
     // themselves, not the underlying files that they point to.
-    let source_inos = sources
+    let source_dev_inos = sources
         .iter()
-        .map(|source| fs::symlink_metadata(source).map(|meta| meta.ino()))
+        .map(|source| fs::symlink_metadata(source).map(|meta| format!("{}/{}", meta.dev().to_string(), meta.ino().to_string())))
         .collect::<Box<_>>();
 
     let mut errors = Vec::new();
 
-    for (ancestor, ino) in dest.ancestors().zip(ancestor_inos) {
-        let ino = ino?;
-        for (source, source_ino) in sources.iter().zip(source_inos.as_ref()) {
-            match source_ino {
-                Ok(source_ino) if *source_ino == ino => errors.push(format!(
+    for (ancestor, dev_ino) in dest.ancestors().zip(ancestor_dev_inos) {
+        let dev_ino = dev_ino?;
+        for (source, source_dev_ino) in sources.iter().zip(source_dev_inos.as_ref()) {
```

Comment:
```suggestion
        for (source, source_id) in sources.iter().zip(source_ids.as_ref()) {
```

Again with the variable name change.